runtime/reference/security_model.md

```diff
@@ -0,0 +1,184 @@
+
+### File system access
+
+By default, executing code can not read or write arbitrary files on the file
+system. This includes listing the contents of directories, checking for the
+existence of a given file, and opening or connecting to Unix sockets.
+
+Some APIs in Deno are implemented using file system operations under the hood,
+even though they do not provide direct read/write access to specific files. Some
+examples of these APIs are:
+
+- `localStorage`
+- Deno KV
+- `caches`
+- `Blob`
+
+Because these APIs are implemented using file system operations, users can use
+them to consume file system resources like storage space, even if they do not
+have direct access to the file system.
+
+During module loading, Deno can load files from disk. This sometimes requires
+explicit permissions, and sometimes is allowed by default:
+
+- All files that are imported from the entrypoint module in a way that they can
+  be statically analyzed are allowed to be read by default. This includes static
+  `import` statements and dynamic `import()` calls where the argument is a
+  string literal that points to a specific file or a directory of files. The
+  full list of files that are in this list can be printed using
+  `deno info <entrypoint>`.
+- Files that are dynamically imported in a way that can not be statically
+  analyzed require runtime read permissions.
+- Files inside of a `node_modules/` directory are allowed to be read by default.
+
+When fetching modules from the network, or when transpiling code from TypeScript
+to JavaScript, Deno uses the file system as a cache. This means that file system
+resources like storage space can be consumed by Deno even if the user has not
+explicitly granted read/write permissions.
+
+### Network access
+
+By default, executing code can not make network requests, open network listeners
+or perform DNS resolution. This includes making HTTP requests, opening TCP/UDP
+sockets, and listening for incoming connections on TCP or UDP.
+
+During module loading, Deno can load modules from the network. By default Deno
+allows loading modules from the following locations using both static and
+dynamic imports:
+
+- `https://deno.land/`
+- `https://jsr.io/`
+- `https://esm.sh/`
+- `https://raw.githubusercontent.com`
+- `https://gist.githubusercontent.com`
+
+These locations are trusted "public good" registries that are not expected to
+enable data exfiltration through URL paths.
+
+In addition Deno allows importing any NPM package through `npm:` specifiers.
+
+Deno also sends requests to `https://dl.deno.land/` at most once a day to check
+for updates to the Deno CLI. This can be disabled using
+`DENO_NO_UPDATE_CHECK=1`.
+
+### Environment variables
+
+By default, executing code can not read or write environment variables. This
+includes reading environment variables, and setting new values.
+
+Deno reads certain environment variables on startup, such as `DENO_DIR` and
+`NO_COLOR` (see `deno help` for the full list).
```

Comment:
Let's link to https://docs.deno.com/runtime/reference/cli/env_variables/ instead